packages/vm/src/evm/evm.ts

```diff
@@ -422,7 +430,36 @@ export default class EVM {
         this._vm._common.isActivatedEIP(3541) &&
         result.returnValue.slice(0, 1).equals(Buffer.from('EF', 'hex'))
       ) {
-        result = { ...result, ...INVALID_BYTECODE_RESULT(message.gasLimit) }
+        if (!this._vm._common.isActivatedEIP(3540)) {
+          result = { ...result, ...INVALID_BYTECODE_RESULT(message.gasLimit) }
+        }
+        // Begin EOF1 contract code checks
+        // EIP-3540 EOF1 header check
+        const eof1CodeAnalysisResults = eof1CodeAnalysis(result.returnValue)
+        if (!eof1CodeAnalysisResults?.code) {
+          result = {
+            ...result,
+            ...INVALID_EOF_RESULT(message.gasLimit),
+          }
+        } else if (this._vm._common.isActivatedEIP(3670)) {
+          // EIP-3670 EOF1 opcode check
+          const codeStart = eof1CodeAnalysisResults.data > 0 ? 10 : 7
+          // The start of the code section of an EOF1 compliant contract will either be
+          // index 7 (if no data section is present) or index 10 (if a data section is present)
+          // in the bytecode of the contract
+          if (
+            !eof1ValidOpcodes(
+              result.returnValue.slice(codeStart, codeStart + eof1CodeAnalysisResults.code)
+            )
```

Comment:
This is from the `EIP-3670` spec:

> At contract creation time both initcode and code are iterated instruction-by-instruction

Might it be that you are doing validation only on the code and not the initcode? Or am I missing something?

Reply:
This is checking the code returned by the create contract to verify its valid .  The initcode is checked when it is run through the `runInterpreter` function call.